directives/duplicate-offscreen.js

```diff
@@ -0,0 +1,78 @@
+import { directive, NodePart } from 'lit-html/lit-html.js';
+import { offscreenStyles } from '../components/offscreen/offscreen.js';
+
+const nestedPartMap = new WeakMap();
+
+/**
+ * Duplicates a node and renders it off-screen
+ *
+ * This is an alteration of the 'duplicate'
+ * directive provided in the lit-html documentation.
+ * https://lit-html.polymer-project.org/guide/creating-directives
+ */
+export const duplicateOffscreen = directive((value) => (containerPart) => {
+	if (!(containerPart instanceof NodePart)) {
+		throw new DirectiveError('duplicateOffscreen directive can only be used in content bindings');
+	}
+
+	let part, copy;
+	const nestedParts = nestedPartMap.get(containerPart);
+	if (nestedParts === undefined) {
+		// duplicate item
+		part = createAndAppend(containerPart);
+		copy = createAndAppend(containerPart);
+		nestedPartMap.set(containerPart, [part, copy]);
+	} else {
+		[ part, copy ] = nestedParts;
+	}
+
+	// set the part
+	part.setValue(value);
+	part.commit();
+
+	// set the copy
+	copy.setValue(value);
+	copy.commit();
+
+	// nextSibling is how we access the actual node we need
+	// This is hardly intuitive and requires knowledge of Part's inner workings
+	// https://github.com/Polymer/lit-html/issues/388
+
+	const copyNode = copy.startNode.nextSibling;
+	copyNode.setAttribute('data-duplicate', '');
+	copyNode.setAttribute('aria-hidden', 'true');
+	copyNode.classList.add('d2l-offscreen');
+
+
+	// create and insert a style tag manually on the shadowRoot
+	// Doing this is actually efficient, as browsers automatically deduplicate
+	// multiple instances of the same style sheet
+	// https://lit-html.polymer-project.org/guide/styling-templates#rendering-in-shadow-dom
+	const shadow = containerPart.options.eventContext.shadowRoot;
+	const style = document.createElement('style');
+
+	style.type = 'text/css';
+	style.appendChild(document.createTextNode(offscreenStyles));
+
+	// workaround for polyfill issue. ShadowRoot isn't a real thing in polyfill land,
+	// so attach the style to the caller instead.
+	try {
+		shadow.prepend(style);
+	} catch (e) {
+		containerPart.options.eventContext.prepend(style);
+	}
```

Comment:
This is a gross "python-ic" way to do this workaround. I would love a suggestion to make this cleaner.

Reply:
What's the error that gets thrown?  Could we avoid this by checking.
```
if (shadow) {
    ...
} else {
    ...
}
```
Also, do you know what `eventContext` is?  The API docs for `NodePart` aren't too informative.  i.e. could the above be accomplished by using `getRootNode()`?  An alternative approach is also to use inline styles and avoid insertion of a stylesheet.